README.md

```diff
@@ -2,36 +2,219 @@
+    chief_config=tfc.MachineConfig(
+            cpu_cores=8,
+            memory=30,
+            accelerator_type=tfc.AcceleratorType.NVIDIA_TESLA_P100,
+            accelerator_count=2),
+    worker_count=0)
+
+datasets, info = tfds.load(name='mnist', with_info=True, as_supervised=True)
+mnist_train, mnist_test = datasets['train'], datasets['test']
+
+num_train_examples = info.splits['train'].num_examples
+num_test_examples = info.splits['test'].num_examples
+
+BUFFER_SIZE = 10000
+BATCH_SIZE = 64
+
+def scale(image, label):
+    image = tf.cast(image, tf.float32)
+    image /= 255
+
+    return image, label
+
+train_dataset = mnist_train.map(scale).cache()
+train_dataset = train_dataset.shuffle(BUFFER_SIZE).batch(BATCH_SIZE)
+
+model = tf.keras.Sequential([
+    tf.keras.layers.Conv2D(32, 3, activation='relu', input_shape=(
+        28, 28, 1)),
+    tf.keras.layers.MaxPooling2D(),
+    tf.keras.layers.Flatten(),
+    tf.keras.layers.Dense(64, activation='relu'),
+    tf.keras.layers.Dense(10, activation='softmax')
+])
+
+model.compile(loss='sparse_categorical_crossentropy',
+              optimizer=tf.keras.optimizers.Adam(),
+              metrics=['accuracy'])
+model.fit(train_dataset, epochs=12)
+```
+
+In this use case, `entry_point` should be `None`. The `run` API can be called anywhere and the entire file will be executed remotely. The API can be called at the end to run the script locally once for debugging purposes (possibly with different #epochs and other flags).
+
+**4. Using `run` within a notebook script that contains the `tf.keras` model.**
+
+![Image of colab](https://github.com/tensorflow/cloud/blob/master/images/colab.png)
+
+In this use case, `entry_point` should be `None` and `docker_image_bucket_name` must be provided.
+
+### What happens when you call run?
+
+The API call will encompass the following:
+1. Making code entities such as a Keras script/notebook, **cloud and distribution ready**.
+2. Converting this distribution entity into a **docker container** with all the required dependencies.
+3. **Deploy** this container at scale and train using Tensorflow distribution strategies.
+4. **Stream logs** and monitor them on hosted TensorBoard, manage checkpoint storage.
+
+By default, we will use local docker daemon for building and publishing docker images to Google container registry. Images are published to `gcr.io/your-gcp-project-id`. If you specify `docker_image_bucket_name`, then we will use [Google cloud build](https://cloud.google.com/cloud-build) to build and publish docker images. 
+
+**Note** If you are using `run` within a notebook script that contains the `tf.keras` model, `docker_image_bucket_name` must be specified.
+
+We use [Google AI platform](https://cloud.google.com/ai-platform/) for deploying docker images on GCP.
+
+Please see `run` API documentation for detailed information on the parameters and how you can control the above processes. 
+
+## End to end examples
```

Comment:
Please add a "constraints & limitations" section describing everything users have to keep in mind that differs from local training.

Reply:
Please also add "Debugging workflow" section explaining how the process users would have to go through to fix any unexpected issues occurring remotely (or otherwise to be able to test locally in conditions that accurately reproduce what happens remotely)

Reply:
Acknowledged. Added the following: will fill in the sections soon.

## Local vs remote training

Things to keep in mind when running your jobs remotely:

[Coming soon]

## Debugging workflow

Here are some tips for fixing unexpected issues occuring remotely.

[Coming soon]

Reply:
Add a `Setup instructions` section as well.